src/Reflex/Dom/Xhr.hs

```diff
@@ -146,11 +150,19 @@ newXMLHttpRequestWithError req cb = do
                    then liftIO $ xmlHttpRequestGetResponseText xhr
                    else  return Nothing
             r <- liftIO $ xmlHttpRequestGetResponse xhr
+            hdrsTxt <- liftIO $ xmlHttpRequestGetAllResponseHeaders xhr
+            let stripBoth (txt1, txt2) = (T.strip txt1, T.strip $ T.drop 1 txt2)
+              -- XMLHttpRequest return all the response headers separated by CRLF
+            let hdrs = Map.fromList
+                  $ fmap (stripBoth . (T.span (/=':')))
+                  $ dropWhileEnd T.null
+                  $ T.splitOn (T.pack "\r\n") hdrsTxt
```

Comment:
Is there an existing header parser we can use?  I'm a bit concerned that there may be edge cases that we haven't thought of.